### ISSUE TITLE
prometheus_server example is broken

### ISSUE THREAD

**Issue by sandrovicente** [metrics-rs/metrics#187]
If we run `prometheus_server` example (` RUST_BACKTRACE=1 cargo run -p metrics-exporter-prometheus --example prometheus_server`), we get:

```
thread 'main' panicked at 'there is no reactor running, must be called from the context of a Tokio 1.x runtime', /home/sandrov/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.4.0/src/runtime/context.rs:18:26
stack backtrace:
   0: rust_begin_unwind
             at /rustc/07e0e2ec268c140e607e1ac7f49f145612d0f597/library/std/src/panicking.rs:493:5
   1: core::panicking::panic_fmt
             at /rustc/07e0e2ec268c140e607e1ac7f49f145612d0f597/library/core/src/panicking.rs:92:14
   2: core::option::expect_failed
             at /rustc/07e0e2ec268c140e607e1ac7f49f145612d0f597/library/core/src/option.rs:1321:5
   3: core::option::Option<T>::expect
             at /home/sandrov/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/option.rs:349:21
   4: tokio::runtime::context::io_handle::{{closure}}
             at /home/sandrov/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.4.0/src/runtime/context.rs:18:13
   5: std::thread::local::LocalKey<T>::try_with
             at /home/sandrov/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/thread/local.rs:272:16
   6: std::thread::local::LocalKey<T>::with
             at /home/sandrov/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/thread/local.rs:248:9
   7: tokio::runtime::context::io_handle
             at /home/sandrov/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.4.0/src/runtime/context.rs:16:9
   8: tokio::io::driver::Handle::current
             at /home/sandrov/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.4.0/src/io/driver/mod.rs:262:13
   9: tokio::io::poll_evented::PollEvented<E>::new_with_interest
             at /home/sandrov/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.4.0/src/io/poll_evented.rs:108:58
  10: tokio::io::poll_evented::PollEvented<E>::new
             at /home/sandrov/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.4.0/src/io/poll_evented.rs:88:9
  11: tokio::net::tcp::listener::TcpListener::from_std
             at /home/sandrov/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.4.0/src/net/tcp/listener.rs:219:18
  12: hyper::server::tcp::AddrIncoming::from_std
             at /home/sandrov/.cargo/registry/src/github.com-1ecc6299db9ec823/hyper-0.14.5/src/server/tcp.rs:38:24
  13: hyper::server::tcp::AddrIncoming::new
             at /home/sandrov/.cargo/registry/src/github.com-1ecc6299db9ec823/hyper-0.14.5/src/server/tcp.rs:30:9
  14: hyper::server::server::Server<hyper::server::tcp::AddrIncoming,()>::try_bind
             at /home/sandrov/.cargo/registry/src/github.com-1ecc6299db9ec823/hyper-0.14.5/src/server/server.rs:74:13
  15: metrics_exporter_prometheus::builder::PrometheusBuilder::build_with_exporter
             at ./metrics-exporter-prometheus/src/builder.rs:221:22
  16: metrics_exporter_prometheus::builder::PrometheusBuilder::install
             at ./metrics-exporter-prometheus/src/builder.rs:159:13
  17: prometheus_server::main
             at ./metrics-exporter-prometheus/examples/prometheus_server.rs:18:5
  18: core::ops::function::FnOnce::call_once
             at /home/sandrov/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ops/function.rs:227:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

Seems that [here](https://github.com/metrics-rs/metrics/blob/3eada5362358a9a70b8f7642c98587b19d72c94a/metrics-exporter-prometheus/src/builder.rs#L158), when we try to enter in the runtime context, the guard is immediately lost.
Should be:

```
diff --git a/metrics-exporter-prometheus/src/builder.rs b/metrics-exporter-prometheus/src/builder.rs
index 82494c2..71be41f 100644
--- a/metrics-exporter-prometheus/src/builder.rs
+++ b/metrics-exporter-prometheus/src/builder.rs
@@ -155,7 +155,7 @@ impl PrometheusBuilder {
             .build()?;

         let (recorder, exporter) = {
-            runtime.enter();
+            let _g = runtime.enter();
             self.build_with_exporter()?
         };
         metrics::set_boxed_recorder(Box::new(recorder))?;
```

Then it runs fine.

**Comment by BratSinot** [comment]
No, everything much simpler =)
No runtime, so we need add runtime:
```
#[tokio::main]
async fn main() {
```

**Comment by tobz** [comment]
The solution proposed by @sandrovicente is the correct one.  While using the `tokio::main` annotation fixes the example, it doesn't fix the `PrometheusBuilder::install` method, which starts its own runtime intentionally.

**Comment by tobz** [comment]
Fixed in #189.

@sandrovicente Thanks for the report!
